misc/pytorch_toolkit/text_recognition/text_recognition/utils/trainer.py

```diff
@@ -340,22 +367,34 @@ def validate(self, use_gt_token=True):
                             pred = ctc_greedy_search(pred, blank_token=self.loss.blank)
                         for j, phrase in enumerate(pred):
                             gold_phrase_str = self.vocab.construct_phrase(
-                                loss_computation_gt[j], ignore_end_token=self.config.get('use_ctc'))
+                                loss_computation_gt[j], ignore_end_token=self.config.get('use_ctc'),
+                                whitespace=self.whitespace)
                             pred_phrase_str = self.vocab.construct_phrase(phrase,
-                                                                          max_len=1 + len(gold_phrase_str.split()),
-                                                                          ignore_end_token=self.config.get('use_ctc')
-                                                                          )
-                            gold_phrase_str = gold_phrase_str.lower()
-                            pred_phrase_str = pred_phrase_str.lower()
+                                                                          max_len=1 + len(gold_phrase_str.split() if self.whitespace else gold_phrase_str),
+                                                                          ignore_end_token=self.config.get('use_ctc'),
+                                                                          whitespace=self.whitespace)
+                            for param in self.config.get('datasets')['validate']:
+                                case_sensitive = param.get('case_sensitive', False)
```

Comment:
```suggestion
                            case_sensitive = getattr(loader.dataset, "case_sensitive", False)
```
In current implementation `case_sensitive` setting is got from the last dataset in the list. Plus, such implementation allows to perform case-sensitive testing for specific dataset (e.g., when you are evaluating on two datasets with different setting `case_sensitive`).